modules/core/src/main/java/org/apache/ignite/internal/processors/query/QuerySchema.java

```diff
@@ -94,17 +97,41 @@ public QuerySchema copy() {
      * @see QuerySchemaPatch
      */
     public QuerySchemaPatch makePatch(Collection<QueryEntity> target) {
+        return makePatch(null, target);
+    }
+
+    /**
+     * Make query schema patch.
+     *
+     * @param targetCfg Cache configuration when it should be changed (enabling indexing dynamically).
+     * @param target Query entity list to which current schema should be expanded.
+     * @return Patch to achieve entity which is a result of merging current one and target.
+     * @see QuerySchemaPatch
+     */
+    public QuerySchemaPatch makePatch(CacheConfiguration<?, ?> targetCfg, Collection<QueryEntity> target) {
         synchronized (mux) {
             Map<String, QueryEntity> localEntities = new HashMap<>();
+            Collection<SchemaAbstractOperation> patchOperations = new ArrayList<>();
+            Collection<QueryEntity> entityToAdd = new ArrayList<>();
+
+            if (entities.isEmpty() && targetCfg != null) {
+                patchOperations.add(new SchemaAddQueryEntityOperation(
+                    UUID.randomUUID(),
+                    targetCfg.getName(),
+                    targetCfg.getSqlSchema(),
+                    target,
+                    targetCfg.getQueryParallelism(),
+                    targetCfg.isSqlEscapeAll()
+                ));
+
+                return new QuerySchemaPatch(patchOperations, entityToAdd, "");
```

Comment:
We can use `Collections.singletonList` here and `Collections.emptyList()` to omit declaration of patchOperations, entityToAdd and localEntities.